RD-1755 Improve user flow spec to be more robust

qooban opened the pull request:
### Changes

1. Changed default timeout to 15s and disabled test retries.

2. Changed user flow spec due to system tests exectution [Stage-UI-System-Test #738](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/738/) which revealed that it does not work with timeout set to 15s. I have improved it a bit so that there are additional checks for blueprint upload to be finished and deployment installation to be started before doing another Cypress actions.

### System tests
1. [Stage-UI-System-Test #740](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/740/pipeline/)
* 7 failures (none of them seems related, see https://github.com/cloudify-cosmo/cloudify-stage/pull/1418#issuecomment-857670188)
2. [Stage-UI-System-Test #742](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/742/pipeline/)
* All test passed
3. [Stage-UI-System-Test #744](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/744/pipeline/)
* 4 failures (none of them seems related, both already reported: RD-2453 + RD-2589)

### To Do

- [x] Update `cloudify-ui-common` after merging https://github.com/cloudify-cosmo/cloudify-ui-common/pull/61 and releasing new version

qooban commented:
Looks like I got (at least partially) similar issues as described recently by @Gelio in https://github.com/cloudify-cosmo/cloudify-stage/pull/1415#issuecomment-856790726 :

1. Maintenance mode button widget should enter maintenance mode on click
2. Snapshots list widget should allow to create and delete a snapshot
3. Snapshots list widget should allow to upload and restore a snapshot
4. Number of nodes widget "before all" hook for "should display nodes count"

I got also:
1. Deployments View widget map should be toggled upon clicking the button (I have to investigate it)
* looks like `POST /searches/deployments` was not called as expected in `useDeploymentsViewWidget` function
2. Deployments View widget bulk actions should allow to create child deployments on filtered deployments
* screenshot looks interesting:
  ![image](https://user-images.githubusercontent.com/5202105/121356929-a42e2480-c931-11eb-81e4-9b40214fa98a.png)
3. Filter refreshes dropdown data on blueprint upload and removal
* looks like RD-2453

I have replayed the job: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/742/pipeline and let's see what we get.

qooban commented:
Hmm... [Stage-UI-System-Test #742](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/742/pipeline) - all test passed.
I have replayed the job again, let's see: [Stage-UI-System-Test #744](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/744/pipeline)

qooban commented:
My conclusions after few more tests (all listed in the description), let's try with timeout set to 15s and no retries (the same configuration as for Cypress). I'll wait for https://github.com/cloudify-cosmo/cloudify-ui-common/pull/61 review and then release new cloudify-ui-common version. Once the new version is available, I'll update this PR and ask for final review.

Gelio commented on test/cypress/support/resource_commons.ts:
```diff
@@ -22,3 +21,7 @@ export function waitUntilEmpty(resource, search, numberOfRetriesLeft = 60, waiti
     });
 }
+
+export function minutesToMs(minutes) {
```
Cool idea to extract this function :+1: 

Minor:

We could add a type annotation :smile: 

```suggestion
export function minutesToMs(minutes: number) {
```

qooban replied:
Right 👍🏻 As the file is `.ts` now, that's good idea to add it.

Gelio reviewed (dismissed):
The code changes look good :+1: Looks like the tests are passing left and right (aside from known errors) - great!

qooban commented:
I updated cloudify-ui-common version and fixed some security vulnerabilities btw (with `npm audit fix`).

Scheduled another system test execution: [Stage-UI-System-Test #747](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/747/pipeline) to do the final check.

qooban commented:
Only agents spec failed:
```
[2021-06-10T11:24:34.338Z]        Spec                                              Tests  Passing  Failing  Pending  Skipped  
[2021-06-10T11:24:34.339Z]   │ ✖  widgets/agents_spec.ts                   01:00        3        -        3        -        - │
...
[2021-06-10T11:24:34.340Z]     ✖  1 of 53 failed (2%)                      27:06      227      224        3        -        -  
```
so as this is known issue (RD-2589), I'm merging this PR 🎉

Gelio commented:
Great :tada:

qooban commented:
> Great tada Now we can fail faster smile

Yes, exactly 🙂 
Quick example: Agents test failure (RD-2589) 
* Nightly: 13m49s
* This PR: 1m00s
🙂
